remoteappmanager/db/orm.py

```diff
@@ -158,10 +159,26 @@ class AppAccounting(ABCAccounting):
     def __init__(self, url, **kwargs):
         self.db = Database(url, **kwargs)
 
+    def check_database_readable(self):
+        ''' Raise IOError if the database url points to a sqlite database
+        that is not readable
+
+        TODO: may extend for validating databases in other dialects?
+        '''
+        db_url = self.db.url
+
+        if db_url.startswith('sqlite:///'):
+            file_path = os.path.abspath(db_url[10:])
+            if not os.access(file_path, os.R_OK):
+                raise IOError(
+                    'Sqlite database {} is not readable'.format(file_path))
+
     def get_user_by_name(self, user_name):
         """ Return an orm.User given a user name.  Return None
         if the user name is not found in the database
         """
+        self.check_database_readable()
```

Comment:
The problem with this solution is that now we are performing this check for every request that is issued. I am not sure if there are other factors in play, but maybe we can just do the check at init, and then assume it's there from then onwards.